File: cash/cash.c
#include <cs50.h> //to get get_float function
#include <math.h> //to get "round" function
#include <stdio.h>

int main(void)
{
    float cash; 
    int cents, coins = 0, coins_added = 0; //declare and initialize variables
    do
    {
        cash = get_float("Change owed: "); //prompts user, sets cash variable
    }
    while (cash < 0); //keep prompting given the condition
    cents = round(cash * 100); //convert dollars into cents
    while (cents / 25 > 0) //check for quarters
    {
        coins_added = cents / 25; 
        cents -= coins_added * 25;
        coins += coins_added;
    }
    while (cents / 10 > 0) //check for dimes
    {
        coins_added = cents / 10;
        cents -= coins_added * 10; 
        coins += coins_added;
    }
    while (cents / 5 > 0) //check for nickels
    {
        coins_added = cents / 5;
        cents -= coins_added * 5;
        coins += coins_added;
    }
    coins_added = cents; //check for pennies
    cents -= coins_added;
    coins += coins_added;
    printf("%i\n", coins); //print least coins given as change
}

File: cash/cash.py
# get get_float from cs50 library
from cs50 import get_float

# list of coin values
coins = [25, 10, 5, 1]
# number of coins
Coins = 0

# condition for user prompt
while True:
    cash = get_float("Change owed: ")
    if cash > 0:
        break
    else:
        True
# convert change to cents
cents = round(cash * 100)
# check if quotient of remaining cents is greater than 0
for i in range(len(coins)):
    if cents / coins[i] > 0:
        coins_added = int(cents / coins[i])
        cents -= int(coins_added * coins[i])
        Coins += int(coins_added)
# print number of coins
print(f"{Coins}")
